### PR TITLE
adds support for mongo paginate pipelines

### PR DESCRIPTION
**Pull request by markbaas**
Had some problems implementing the tests. Perhaps you could give me some instructions here.

### CODE REVIEW

**Comment by zsltg** [comment]
> 
> 
> Had some problems implementing the tests. Perhaps you could give me some instructions here.

I haven't looked into this in detail (@uriyyo please correct me if I am wrong), but I believe with the current test setup, instead of including both in the same file, you would need to create a separate one and include the tests for the aggregate there, eg. `test_motor_aggregate.py`.

You can try to run it locally and see if it works:

`docker run -p 27017:27017 --name mongo -d mongo:5.0.3`
`poetry run pytest tests/ext/test_motor_aggregate.py --cov-report=xml --ignore=tests/ext/test_async_sqlalchemy.py --ignore=tests/ext/test_sqlalchemy_future.py --ignore=tests/ext/test_sqlmodel.py`

You can also run the whole test workflow with [act](https://github.com/nektos/act), the test cases pass, only the Codecov upload will fail.

**Comment by uriyyo** [comment]
@zsltg Thanks, for the detailed explanation.

@markbaas Please, follow `Zsolt` instruction.

File: tests/ext/test_motor_aggregate.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    @app.on_event("startup")
+    async def on_startup() -> None:
+        for _ in range(100):
+            await db_client.test.users.insert_one({"name": faker.name()})
+
+    @app.on_event("shutdown")
+    async def on_shutdown() -> None:
+        # await db_client.test.users.delete_many({})
+        # db_client.drop_database("test")
```
Comment (uriyyo) [inline tests/ext/test_motor_aggregate.py:35]: Please, remove commented code.

File: tests/ext/test_motor_aggregate.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    async def client(self, app):
+        with SafeTestClient(app) as c:
+            yield c
+
+    @fixture(scope="session")
+    async def entities(self, db_client):
+        cursor = db_client.test.users.aggregate(
+            [{"$group": {"_id": "$name", "name": {"$first": "$name"}}}, {"$sort": {"name": 1}}]
+        )
+        # cursor = db_client.test.users.find()
```
Comment (uriyyo) [inline tests/ext/test_motor_aggregate.py:61]: Please, remove commented code.

File: tests/ext/test_motor_aggregate.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        items = await cursor.to_list(length=None)
+        # return [{"name": x["name"]} for x in items]
```
Comment (uriyyo) [inline tests/ext/test_motor_aggregate.py:63]: Please, remove commented code.

**Review by uriyyo — CHANGES_REQUESTED** [review]
LGTM in general.

Only a few minor things, please remove commented code and run pre-commit hooks, because the lint stage is failing.

To run pre-commit hooks locally you can run the following command:
```sh
poetry run pre-commit run --all-files
```

**Comment by uriyyo** [comment]
@markbaas Do you have plans to update this PR?

**Comment by markbaas** [comment]
> @markbaas Do you have plans to update this PR?

sorry done!

**Review by uriyyo — APPROVED** [review]
LGTM

**Pull request merged**